lib/src/base/wave_style.dart

```diff
@@ -1,77 +1,90 @@
+  /// Space between each wave
   final double spacing;
 
-  ///Whether to show upper wave or not defaults to true
+  /// Whether to show upper wave or not defaults to true
   final bool showTop;
 
-  ///Whether to show bottom wave or not default to true
+  /// Whether to show bottom wave or not default to true
   final bool showBottom;
 
-  ///Wave padding from bottom default is size.height/2
+  /// Wave padding from bottom default is size.height/2
   final double? bottomPadding;
 
-  ///The kind of finish to place on the end of lines drawn
-  /// default to StrokeCap.round
+  /// The kind of finish to place on the end of lines drawn
+  ///  default to StrokeCap.round
   final StrokeCap waveCap;
 
-  ///Color line in the middle
+  /// Color line in the middle
   final Color middleLineColor;
 
-  ///Thickness of middle line.
+  /// Thickness of middle line.
   final double middleLineThickness;
 
-  ///Width of each wave
+  /// Width of each wave
   final double waveThickness;
 
-  ///The background color of waveform box default is Black
+  /// The background color of waveform box default is Black
   final Color backgroundColor;
 
-  ///Extend the wave to the end of size.width, default is size.width/2
+  /// Extend the wave to the end of size.width, default is size.width/2
   final bool extendWaveform;
 
-  ///Show duration labels. Default is false
+  /// Show duration labels. Default is false
   final bool showDurationLabel;
 
-  ///Show duration label in HH:MM:SS format. Default is MM:SS
+  /// Show duration label in HH:MM:SS format. Default is MM:SS
   final bool showHourInDuration;
 
-  ///Text style for duration labels
+  /// Text style for duration labels
   final TextStyle durationStyle;
 
-  ///Color of duration lines
+  /// Color of duration lines
   final Color durationLinesColor;
 
-  ///Height of duration lines
+  /// Height of duration lines
   final double durationLinesHeight;
 
-  ///Space between duration labels and waveform square
+  /// Space between duration labels and waveform square
   final double labelSpacing;
 
-  ///It might happen that label text gets cut or have extra clipping.
+  /// It might happen that label text gets cut or have extra clipping.
   ///
-  ///So use this to add or remove clipping
+  /// So provided +Ve value add more clipping and -Ve will reduce
+  /// the clipping.
   final double? extraClipperHeight;
 
-  ///Value > 0 will be padded right and value < 0 will be padded left
+  /// Value > 0 will be padded right and value < 0 will be padded left.
   final double durationTextPadding;
 
-  ///Provide gradient to waveform using this. Use shader as shown in example.
+  /// Applies this gradient to waveforms.
+  ///
+  /// Use as below,
+  ///
+  /// import 'dart:ui' as ui show Gradient;
+  ///
+  /// ...
+  ///
+  /// ui.Gradient.linear(
+  ///                 const Offset(70, 50),
+  ///                 Offset(MediaQuery.of(context).size.width / 2, 0),
+  ///                 [Colors.red, Colors.green],
```

Comment:
any reason for living commented code?

Reply:
This is an example code snippet for showing how to use shader while applying gradient